### PR TITLE
Added delete and update tasks

### PR DESCRIPTION
**Pull request by MkpYurevaLyubov**

### CODE REVIEW

File: src/Controllers/todoList.controller.js [hunk 1]
```diff
@@ -20,7 +20,32 @@ const createTask = (req, res) => {
...
+  if (!req.body) return res.sendStatus(400);
+
+  const id = req.body._id;
+  const newTask = {
+    text: req.body.text,
+    isCheck: req.body.isCheck
+  };
+
+  Task.findOneAndUpdate({ _id: id }, newTask, {new: true}, (err, task) => {
+    if (err) return console.log(err);
```
Comment (omanei) [inline src/Controllers/todoList.controller.js]: remove console.log
send error to client

File: src/Controllers/todoList.controller.js [hunk 2]
```diff
@@ -20,7 +20,32 @@ const createTask = (req, res) => {
   });
 };
 
+const updateTask = (req, res) => {
+  if (!req.body) return res.sendStatus(400);
+
+  const id = req.body._id;
```
Comment (omanei) [inline src/Controllers/todoList.controller.js:25]: check id maybe he is empty or not exist

File: src/Controllers/todoList.controller.js [hunk 3]
```diff
@@ -20,7 +20,32 @@ const createTask = (req, res) => {
...
+  };
+
+  Task.findOneAndUpdate({ _id: id }, newTask, {new: true}, (err, task) => {
+    if (err) return console.log(err);
+    res.send(task);
+  });
+};
+
+const deleteTask = (req, res) => {
+  const id = req.body._id;
```
Comment (omanei) [inline src/Controllers/todoList.controller.js]: same as above

File: src/Controllers/todoList.controller.js [hunk 4]
```diff
@@ -20,7 +20,32 @@ const createTask = (req, res) => {
...
+  Task.findByIdAndDelete(id, (err, task) => {
+    if (err) return console.log(err);
```
Comment (omanei) [inline src/Controllers/todoList.controller.js]: same as above

File: src/Controllers/todoList.controller.js [hunk 5]
```diff
@@ -21,24 +22,27 @@ const createTask = (req, res) => {
 };
 
 const updateTask = (req, res) => {
-  if (!req.body) return res.sendStatus(400);
-
+  const isEmpty = Object.keys(req.body).length === 0;
```
Comment (omanei) [inline src/Controllers/todoList.controller.js]: check all body keys

File: src/Controllers/todoList.controller.js [hunk 6]
```diff
@@ -2,17 +2,20 @@ const Task = require("../Models/todoList.models");
...
-  if (!req.body.text || !req.body.isCheck) return res.sendStatus(400);
+  const body = req.body;
+  if (!body.hasOwnProperty('text')) return res.status(404).send("Error! Text not found");
+  if (!body.hasOwnProperty('isCheck')) return res.status(404).send("Error! Checkbox not found");
+  if (typeof body.isCheck !== "boolean") return res.status(422).send("Error! Checkbox value not correct");
 
   const task = new Task({
-    text: req.body.text,
-    isCheck: req.body.isCheck
+    text: body.text,
```
Comment (DoctorFreed) [inline src/Controllers/todoList.controller.js:17]: send only body

File: src/Controllers/todoList.controller.js [hunk 7]
```diff
@@ -21,26 +24,31 @@ const createTask = (req, res) => {
...
+  if (!body.hasOwnProperty('text')) return res.status(404).send("Error! Text not found");
+  if (!body.hasOwnProperty('isCheck')) return res.status(404).send("Error! Checkbox not found");
+  if (!body.hasOwnProperty('_id')) return res.status(404).send("Error! Id not found");
+  if (typeof body.isCheck !== "boolean") return res.status(422).send("Error! Checkbox value not correct");
+  if (typeof id !== "string") return res.status(422).send("Error! Id value not correct");
 
   const newTask = {
-    text: req.body.text,
-    isCheck: req.body.isCheck
+    text: body.text,
```
Comment (DoctorFreed) [inline src/Controllers/todoList.controller.js:36]: send only body

File: src/Controllers/todoList.controller.js (hunk 6 above)
Comment (ghost) [inline src/Controllers/todoList.controller.js]: only body

File: src/Controllers/todoList.controller.js (hunk 7 above)
Comment (ghost) [inline src/Controllers/todoList.controller.js]: only body

**Pull request merged**